frontend/src/redux/reducers/ui.tsx

```diff
@@ -220,6 +221,11 @@ function reducer(state = _.cloneDeep(INITIAL_STATE), action: Action) {
       newFilters.notifications.sort((n1: Notification, n2: Notification) => {
         return new Date(n2.date).getTime() - new Date(n1.date).getTime();
       });
+      // limit showing not more then max_notification_count notifications for now till we have a dedicated page for notifications
+      if (newFilters.notifications.length >= max_notification_count) {
+        // we remove the oldest available notifications when the notifications length is crossing max_notification_count threshold
+        newFilters.notifications.pop();
```

Comment:
I understand this should work, but I am getting the UI to be unusable still with this change.
It can be that localStorage in my tests still had loads of notifications and that was getting loaded...
I noticed that limiting the length earlier does work for me. i.e. having the following above:
```
[...notifications].slice(0, max_notification_count).concat(newFilters.notifications),
          function (item) {
            return item.id;
          }
```